include/klee/Core/TerminationTypes.h

```diff
@@ -0,0 +1,138 @@
+//===-- TerminationTypes.h --------------------------------------*- C++ -*-===//
+//
+//                     The KLEE Symbolic Virtual Machine
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef KLEE_TERMINATIONTYPES_H
+#define KLEE_TERMINATIONTYPES_H
+
+#include <cstdint>
+
+
+///@brief Categorises StateTerminationType.
+enum class StateTerminationClass : std::uint8_t {
+  NONE = 0, ///< (do not write tests)
+  Exit,     ///< normal program exit
+  Early,    ///< early termination
+  ProgErr,  ///< program errors
+  UserErr,  ///< user errors
+  ExecErr,  ///< execution errors
+};
+
+
+///@brief Reason an ExecutionState got terminated.
+enum class StateTerminationType : std::uint32_t {
```

Comment:
This is such an annoying pattern, is there no nicer way of doing it, where you only need to add the type in one place? Otherwise it might be nice to have some sort of "Cheat" type that you can use when others don't fit.

Reply:
Not sure what you mean - a full-blown class with casts etc.?

Reply:
Yeah something like that. Not sure if you need casts.

You could have global instantiations of a class instead of enums (and hide the constructor somehow)

Reply:
That "somehow" might actually be the problem without macros...

Reply:
Shouldn't a private constructor be enough? Then you can have public static instances of the object and use `using namespace` to use them as enums?